src/cpu/kernel/assembler.rs

```diff
@@ -41,6 +41,28 @@ pub fn segment_kernel(
     }
 }
 
+pub fn segment_kernel_with_data(
+    basedir: &str,
+    block: &str,
+    file: &str,
+    seg_file_data: Vec<u8>,
```

Comment:
Can we use io.Read here? so we don't have to implement 2 functions.

Reply:
Done.